source/ota.c

```diff
@@ -86,182 +95,386 @@ typedef struct OtaStateTableEntry
+ */
 static IngestResult_t processDataBlock( OtaFileContext_t * pFileContext,
                                         uint32_t uBlockIndex,
                                         uint32_t uBlockSize,
                                         OtaPalStatus_t * pCloseResult,
                                         uint8_t * pPayload );
 
-/* Free the resources allocated for data ingestion and close the file handle. */
+/**
+ * @brief Free the resources allocated for data ingestion and close the file handle.
+ *
+ * @param pFileContext Information of file to be streamed.
+ * @param pCloseResult Result of closing file in PAL.
+ * @return IngestResult_t IngestResultAccepted_Continue if successful, other error for failure.
+ */
 
 static IngestResult_t ingestDataBlockCleanup( OtaFileContext_t * pFileContext,
                                               OtaPalStatus_t * pCloseResult );
 
-/* Called to update the filecontext structure from the job. */
-
+/**
+ * @brief Get the File Context From Job Document.
+ *
+ * We received an OTA update job message from the job service so process
+ * the message and update the file context.
+ *
+ * @param pRawMsg Raw job document.
+ * @param messageLength length of document.
+ * @return OtaFileContext_t* Information of file to be streamed.
+ */
 static OtaFileContext_t * getFileContextFromJob( const char * pRawMsg,
                                                  uint32_t messageLength );
 
-/* Validate JSON document */
+/**
+ * @brief Validate JSON document and the DocModel.
+ *
+ * @param pJson JSON job document.
+ * @param messageLength  Length of the job document.
+ * @return DocParseErr_t DocParseErrNone if successful, JSON document parser errors.
+ */
 
 static DocParseErr_t validateJSON( const char * pJson,
                                    uint32_t messageLength );
 
-/* Store the parameter from the json to the offset specified by the document model. */
-
+/**
+ * @brief Store the parameter from the json to the offset specified by the document model.
+ *
+ * @param docParam Structure to store the details of keys and where to store them.
+ * @param pContextBase Start of file context.
+ * @param pValueInJson Pointer to the value of the key in JSON buffer.
+ * @param valueLength Length of the value.
+ * @return DocParseErr_t DocParseErrNone if successful, JSON document parser errors.
+ */
 static DocParseErr_t extractParameter( JsonDocParam_t docParam,
                                        void * pContextBase,
                                        const char * pValueInJson,
                                        size_t valueLength );
 
-/* Parse a JSON document using the specified document model. */
-
+/**
+ * @brief Extract the desired fields from the JSON document based on the specified document model.
+ *
+ * @param pJson JSON job document.
+ * @param messageLength  Length of the job document.
+ * @param pDocModel Details of expected parameters in the job doc.
```

Comment:
Are we not defining whether these params are [in|out|inout]? Out of curiosity.

Reply:
Good callout, missed that,
Addressed in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/116/commits/3e5b1d96211d212a79fd5aa4a1bd33a51add1e85